File: integration-tests/hibernate-orm-panache/src/test/java/io/quarkus/it/panache/MockPersonRepository.java
package io.quarkus.it.panache;

import jakarta.annotation.Priority;
import jakarta.enterprise.context.ApplicationScoped;
import jakarta.enterprise.inject.Alternative;

/**
 * An alternate implementation of {@link PersonRepository} to demonstrate that Repository entity type can be retrieved
 * even when the class does not directly implement PanacheRepository interface.
 */
@Alternative
@Priority(1)
@ApplicationScoped
public class MockPersonRepository extends PersonRepository {
}

File: integration-tests/hibernate-orm-panache-kotlin/src/test/kotlin/io/quarkus/it/panache/MockPersonRepository.kt
package io.quarkus.it.panache

import io.quarkus.it.panache.kotlin.PersonRepository
import jakarta.annotation.Priority
import jakarta.enterprise.context.ApplicationScoped
import jakarta.enterprise.inject.Alternative

/**
 * An alternate implementation of [PersonRepository] to demonstrate that Repository entity type can be retrieved
 * even when the class does not directly implement PanacheRepository interface.
 */
@Alternative
@Priority(1)
@ApplicationScoped
open class MockPersonRepository : PersonRepository()
